src/api/app/models/user.rb

```diff
@@ -127,6 +128,18 @@ def create_home_project
     self.login_failure_count = 0 if login_failure_count.nil?
   end
 
+  # autocomplete method
+  # return either an array of users or an array of hashes
+  def self.autocomplete(prefix = nil, is_token = false)
+    User.where('login LIKE ?', "#{prefix}%").pluck(:login).collect do |user|
```

Comment:
You don't need to put the `User` class in front as you are already in the `User` model

Reply:
What if you create a scope out of `where('login LIKE ?', "#{prefix}%")`